src/scoped.component.js

```diff
@@ -9,21 +9,35 @@ const reactSupportsReturningArrays = !!ReactDOM.createPortal;
 
 export class Scoped extends React.Component {
   static propTypes = {
-    css: PropTypes.string.isRequired,
+    css: PropTypes.string,
+    postcss: PropTypes.object,
     namespace: PropTypes.string,
   }
 
   static defaultNamespace = 'kremling'
 
   constructor(props) {
     super(props);
-    this.state = this.newCssState(props);
+    if (!props.css && !props.postcss) throw Error(`Kremling's <Scoped /> component requires either the 'css' or 'postcss' props.`);
+    if (props.css && props.postcss) throw Error(`Kremling's <Scoped /> component requires either the 'css' or 'postcss' props. Cannot use both.`);
+    if (props.postcss && !(props.postcss.styles && props.postcss.id)) throw Error(`Kremlings's <Scoped /> component 'postcss' prop requires an object containing 'styles' and 'id' properties. Try using the kremling-loader.`);
+    if (props.css) {
+      this.state = this.newCssState(props);
+    } else {
+      this.state = this.newPostcssState(props);
+    }
   }
 
   render() {
     const kremlingChildren = React.Children.map(this.props.children, child => {
       if (React.isValidElement(child)) {
-        return React.cloneElement(child, {[this.state.kremlingAttrName]: this.state.kremlingAttrValue});
+        let elProps;
+        if (this.props.css) {
+          elProps = { [this.state.kremlingAttrName]: this.state.kremlingAttrValue };
+        } else {
+          elProps = { className: `${this.props.postcss.id} ${child.props && child.props.className ? child.props.className : ''}` };
```

Comment:
since we're using classes now instead of data attributes, we need to add our new scoped class along with the child's className prop

Reply:
Could you help me understand this? I at first was thinking that the postcss id was just for preventing the duplication of the same style element. But now it looks like it's used as part of our implementation of scoping? I'm not sure if that will work.